src/library/Uno.Material/Styles/Controls/v2/CheckBox.xaml

```diff
@@ -755,23 +530,35 @@
 							<ColumnDefinition Width="*" />
 						</Grid.ColumnDefinitions>
 
-						<Ellipse x:Name="FocusRing"
+						<Ellipse x:Name="UncheckedFocusedCircle"
+								 HorizontalAlignment="Center"
+								 VerticalAlignment="Center"
+								 Stretch="Uniform"
+								 Width="{ThemeResource CheckBoxFocusAreaSize}"
+								 Height="{ThemeResource CheckBoxFocusAreaSize}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillUnchecked}"
+								 Visibility="{Binding IsChecked, RelativeSource={RelativeSource Mode=TemplatedParent}, Converter={StaticResource CheckBoxIsCheckedToFocusCircleCollapsed}, TargetNullValue=Collapsed, FallbackValue=Collapsed}"
+								 IsHitTestVisible="False"
+								 Opacity="0" />
+
+						<Ellipse x:Name="CheckedOrIndeterminateFocusedCircle"
 								 HorizontalAlignment="Center"
 								 VerticalAlignment="Center"
 								 Stretch="Uniform"
 								 Width="{ThemeResource CheckBoxFocusAreaSize}"
 								 Height="{ThemeResource CheckBoxFocusAreaSize}"
-								 Fill="{TemplateBinding Background}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillChecked}"
+								 Visibility="{Binding IsChecked, RelativeSource={RelativeSource Mode=TemplatedParent}, Converter={StaticResource CheckBoxIsCheckedToFocusCircleVisible}, TargetNullValue=Collapsed, FallbackValue=Collapsed}"
 								 IsHitTestVisible="False"
 								 Opacity="0" />
 
-						<Ellipse x:Name="PressRing"
+						<Ellipse x:Name="HoverAndPressedCircle"
 								 HorizontalAlignment="Center"
 								 VerticalAlignment="Center"
 								 Stretch="Uniform"
 								 Width="{ThemeResource CheckBoxFocusAreaSize}"
 								 Height="{ThemeResource CheckBoxFocusAreaSize}"
-								 Fill="{TemplateBinding Background}"
+								 Fill="{ThemeResource CheckBoxStatesCircleFillChecked}"
 								 IsHitTestVisible="False"
 								 Opacity="0" />
```

Comment:
The naming is bothering me, I wish we could simplify it a bit. 

In the Lightweight Styling keys its called a "StatesCircle"

> CheckBox**StatesCircle**FillUnchecked
> CheckBox**StatesCircle**FillChecked

But the actual Ellipses are just called "Circle"

> UncheckedFocused**Circle**
> CheckedOrIndeterminateFocused**Circle**
> HoverAndPressed**Circle**

I kind of liked the "Ring" naming, so could it just be a StateRing?

> CheckBox**StateRing**FillUnchecked
> CheckBox**StateRing**FillChecked

> UncheckedFocused**StateRing**
> CheckedOrIndeterminateFocused**StateRing**
> HoverAndPressed**StateRing**


Now for the naming of the Circles themselves. We differentiate between "Unchecked" and "CheckedOrIndeterminate" but there only exists a FillUnchecked and FillChecked, no FillIndeterminate for the StatesCircle resource keys. Meaning we wouldn't be able to customize the circle between Indeterminate and Checked. In that case, could we just renaming the Ellipses to UncheckedFocusStateRing and CheckedFocusStateRing? And maybe instead of HoverAndPressedStateRing it would be PointerStateRing?

Reply:
Also, this state ring is specific to the Material design guidelines so should we prefix these with `Material`?

Reply:
thanks a lot for the feedback @kazo0 :) Really appreciated and I was not 100% on board with these long names but had a hard time coming up with something that was better at the time (more focused on the style + visual itself)

So let's recap your feedback:
- For the Ring part, I wanted something else as a Ring it is a circle with an empty circle inside (similar to a donut), and here for all the visual states it is a filled circle (so we are losing the "Ring" notion)
![image](https://github.com/unoplatform/Uno.Themes/assets/16295702/9869f24b-80e6-4b88-ad91-6c39dff8dce7)
So wasn't sure on keeping the "Ring" name here

- I agree "PointerState" sounds better than "HoverAndPressedState"

- For the "UncheckedFocused" and "CheckedOrIndeterminateFocused" yes would be better to have "UncheckedFocused", "CheckedFocused"

Reply:
Maybe Circle instead of Ring then, something like this pattern

Resource keys

CheckBoxStateCircleFillUnchecked
CheckBoxStateCircleFillChecked

x:Names

UncheckedFocusStateCircle
CheckedFocusStateCircle
PointerStateCircle

Reply:
@kazo0 code updated with similar naming pattern across CheckBox, RadioButton, and ToggleButton (all 3 are using similar StateCircle for their visual states).
Let me know if everything is fine now or if something else needs an update. 
Thanks again for your feedback !